### PR TITLE
Confine autofs_version fact to Linux kernel.

### LINKED ISSUE #68: autofs_version fact throws exception on mac OS X
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 4.8.1
- Ruby: ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin15]
- Distribution: El Capitan
- Module version: 3.0.1, 3.0.2 and presumably 3.0.0

## How to reproduce (e.g Puppet code you use)

Run puppet apply with any puppet manifests. Don't need to be using autofs resources, just need to have puppet/autofs in your Puppetfile (and have done a pluginsync).

## What are you seeing

```
 DEBUG [71b964ab] 	Debug: Facter: searching for custom facts in /opt/puppetlabs/puppet/cache/lib/facter.
 DEBUG [71b964ab] 	Debug: Facter: fact "agent_specified_environment" resolved to null and will not be added.
 DEBUG [71b964ab] 	Debug: Facter: fact "apt_has_updates" resolved to null and will not be added.
 DEBUG [71b964ab] 	Debug: Facter: fact "apt_package_updates" resolved to null and will not be added.
 DEBUG [71b964ab] 	Debug: Facter: fact "apt_reboot_required" resolved to null and will not be added.
 DEBUG [71b964ab] 	Debug: Facter: fact "apt_security_updates" resolved to null and will not be added.
 DEBUG [71b964ab] 	Debug: Facter: fact "apt_update_last_success" resolved to null and will not be added.
 DEBUG [71b964ab] 	Debug: Facter: fact "apt_updates" resolved to null and will not be added.
 DEBUG [71b964ab] 	Debug: Facter: fact "archive_windir" resolved to null and will not be added.
 DEBUG [71b964ab] 	Debug: Facter: executing command: /bin/sh -c /usr/sbin/automount -V 2>&1
 DEBUG [71b964ab] 	Debug: Facter: /usr/sbin/automount: illegal option -- V
 DEBUG [71b964ab] 	Debug: Facter: process exited with status code 1.
 DEBUG [71b964ab] 	Debug: Facter: process exited with status code 1.
 DEBUG [71b964ab] 	Error: Facter: error while resolving custom fact "autofs_version": undefined method `[]' for nil:NilClass
```

## What behaviour did you expect instead

No crash, and autofs_version fact returning something like nil or empty string.

## Any additional information you'd like to impart

This new autofs_version fact was added in this PR:

https://github.com/voxpupuli/puppet-autofs/pull/33

### PR DESCRIPTION
**Pull request by jgreen210**
Fixes #68

https://github.com/voxpupuli/puppet-autofs/issues/68

### CODE REVIEW

**Comment by jgreen210** [comment]
Closing this PR since still get an error:

```
 DEBUG [6b42cc02] 	Debug: Facter: executing command: /bin/sh -c /usr/sbin/automount -V 2>&1
 DEBUG [6b42cc02] 	Debug: Facter: /usr/sbin/automount: illegal option -- V
 DEBUG [6b42cc02] 	Debug: Facter: process exited with status code 1.
 DEBUG [6b42cc02] 	Debug: Facter: process exited with status code 1.
 DEBUG [6b42cc02] 	Error: Facter: error while resolving custom fact "autofs_version": undefined method `[]' for nil:NilClass
 DEBUG [6b42cc02] 	Debug: Facter: fact "autofs_version" resolved to null and will not be added.
```

I can look at this some more next week.

**Comment by jgreen210** [comment]
Reopening the PR - the changes do work.

It looks like librarian-puppet doesn't handle moving between puppet forge and git nicely, so I wasn't testing the right version of the code on Friday.

File: lib/facter/autofs.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 Facter.add(:autofs_version) do
+  confine kernel: 'Linux'
```
Comment (alexjfisher) [inline lib/facter/autofs.rb]: Solaris (an OS this module provides some support for) doesn't support `automount -V` either.
```
afisher@solaris:~$ automount -V
automount: illegal option -- V
automount: Usage: automount  [ -v ]  [ -t duration ]
```

**Comment by alexjfisher** [comment]
Thanks @jgreen210 !

**Pull request merged**